astropy/cosmology/funcs.py

```diff
@@ -113,6 +151,16 @@ def z_at_value(func, fval, zmin=1e-8, zmax=1000, ztol=1e-8, maxfun=500):
     other commonly inverted quantities) are monotonic in flat and open
     universes, but not in closed universes.
     """
+    is_array = (
+           (isinstance(fval, Quantity) and not fval.isscalar) or
+           (not isinstance(fval, Quantity) and isinstance(fval, np.ndarray))
+           )
```

Comment:
```suggestion
    is_array = (isinstance(fval, Quantity) and not fval.isscalar) or np.array(fval).ndim > 0
```
Should be mostly equivalent, in fact this would identify any iterable (array_like) `fval`.